nucypher/blockchain/eth/actors.py

```diff
@@ -292,13 +298,26 @@ def connect_condition_providers(
                     raise self.ActorError(
                         f"Condition blockchain endpoint {uri} is not on chain {chain_id}"
                     )
-                providers[int(chain_id)].add(provider)
+                healthy = rpc_endpoint_health_check(endpoint=uri)
+                if not healthy:
+                    self.log.warn(
+                        f"user-supplied condition RPC endpoint {uri} is unhealthy"
+                    )
+                providers[int(chain_id)].append(provider)
+
+        # Ingest default/fallback RPC providers for each chain
+        for chain_id in self.domain.condition_chain_ids:
+            default_endpoints = get_healthy_default_rpc_endpoints(chain_id)
+            for uri in default_endpoints:
+                provider = self._make_condition_provider(uri)
+                providers[chain_id].append(provider)
```

Comment:
After considering this, I don't think it poses any concern.  I will however, cast this list to an intermediate set to de-duplicate.

Reply:
Ended up implementing a strategy that preserves order while still de-duplicating ✔️

Reply:
You are correct that duplicates can exist within the user-supplied endpoints, but duplicates can also exist across user-supplied endpoints *and* fallback endpoints, e.g., if the user-supplied endpoint value was a public RPC endpoint that is also in the fallback list.

For example, supposed in the ursula.json file the user's endpoint for sepolia is `https://rpc.sepolia.org`, that is the same endpoint that is included in the fallback list, hence the duplication issue.

Here is a commit that addresses what I meant - https://github.com/derekpierre/nucypher/commit/9ca2a2a042eda7e39a39c629991ab7055236e64e. Take a look. You can cherry-pick it or do your own implementation as deemed necessary.

Reply:
Ah, Gotcha.  Well, changing datatypes from a set to a list removed duplicate protection for user-supplied endpoints but the case you describe, where a user has supplied an endpoint that is already in the fallback list is a pretty unusual/rare corner case in my opinion.  In fact, it's not completely clear what the ideal behavior or the user's intentions is in this situation 🤔 perhaps we just issue a logging warning? Is this problematic somehow?  Perhaps they intend to prioritize this endpoint?

Reply:
> it's not completely clear what the ideal behavior or the user's intentions is in this situation 🤔 perhaps we just issue a logging warning? 

I think we prioritize the user-supplied endpoint, as we already do with our logic; whatever endpoint they provide. Agreed on the warning - the commit logs a warning for that situation.

> Is this problematic somehow?
It's more of a potential efficiency issue. If the endpoint ends up not being healthy down the road, we may try it twice (and it fail twice) during condition evaluation because of the dupe. This is definitely an edge case, though, but it is a possibility.

Reply:
Your implementation looks workable - but I'm unsure that ignoring a duplicate endpoint is the desired behavior here?  What do you think about crashing instead?  or just permitting the duplicate but issuing a warning?